### PR TITLE
cmd,skip: add flag to skip test under race

### PR DESCRIPTION
**Pull request by rytaft**
This commit updates the `skip-test` command to support a flag `-under_race`,
which only skips the test under race builds. This commit also adds a new
function to the skip package, `UnderRaceWithIssue`, which makes it easier to
skip a test under race and log the associated issue.

Release note: None

### CODE REVIEW

**Review by jordanlewis — APPROVED** [review]
LGTM, thanks!

File: pkg/testutils/skip/skip.go [hunk 1]
```diff
@@ -49,6 +49,16 @@ func UnderRace(t SkippableTest, args ...interface{}) {
 	}
 }
 
+// UnderRaceWithIssue skips this test if the race detector is enabled,
+// logging the given issue ID as the reason.
+func UnderRaceWithIssue(t SkippableTest, githubIssueID int, args ...interface{}) {
+	if util.RaceEnabled {
+		t.Skip(append([]interface{}{fmt.Sprintf(
```
Comment (otan) [inline pkg/testutils/skip/skip.go:56]: should this just be

```
t.Skip(fmt.Sprintf("disabled under race. issue: https://github.com/cockroachdb/cockroach/issue/%d", append([]interface{}{githubIssueID}, args...)...)
```

**Review by rytaft — COMMENTED** [review]
TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55857#-:-MKGz5g12iidCzf72DcI:bz5mair)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/testutils/skip/skip.go, line 56 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55857#-MKFioKJtBVMvy4SLSk9-r1-56:-MKGyBb0AYAB5Q9owvjo:bkpw722) ([raw file](https://github.com/cockroachdb/cockroach/blob/ea443db7e7675f3a6a9475a998e21aa154ce760f/pkg/testutils/skip/skip.go#L56)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

should this be

```
t.Skip(fmt.Sprintf(
   "disabled under race. issue: https://github.com/cockroachdb/cockroach/issue/%d",
   append([]interface{}{githubIssueID}, args...)...,
))
```

(i think the `[]interface{}` should be nested inside the sprintf)
</blockquote></details>

I don't think so... we want to pass the `[]interface{}` to `t.Skip` so it can use the first item for the `Skip` string and the remaining items for the `SkipDetails` string. At least that's what the other functions here seem to be doing...

**Review by otan — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55857)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rytaft)

___
*[pkg/testutils/skip/skip.go, line 56 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55857#-MKFioKJtBVMvy4SLSk9-r1-56:-MKH0a5o2ForTErljA3S:b-5ynvt) ([raw file](https://github.com/cockroachdb/cockroach/blob/ea443db7e7675f3a6a9475a998e21aa154ce760f/pkg/testutils/skip/skip.go#L56)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

I don't think so... we want to pass the `[]interface{}` to `t.Skip` so it can use the first item for the `Skip` string and the remaining items for the `SkipDetails` string. At least that's what the other functions here seem to be doing...
</blockquote></details>

oh, i see!

**Comment by rytaft** [comment]
TFTRs!

bors r+

**Pull request merged**